pymatgen_diffusion/neb/full_path_mapper.py

```diff
@@ -129,61 +161,233 @@ def _get_pos_and_migration_path(self, u, v, w):
         """
+
         for u, v, d in self.s_graph.graph.edges(data=True):
-            if d['hop_label'] == taget_label:
+            if d["hop_label"] == target_label:
                 d.update(data)
+                if m_path is not None:
+                    # Try to override the data.
+                    if not m_path.symm_structure.spacegroup.are_symmetrically_equivalent(
+                        [m_path.isite], [d["hop"].isite]
+                    ):
+                        # "The data going to this edge needs to be flipped"
+                        for k in data.keys():
+                            if isinstance(data[k], (np.ndarray, np.generic)):
+                                raise Warning(
+                                    "The data provided will only be flipped "
+                                    "if it a list"
+                                )
+                            if not isinstance(data[k], list):
+                                continue
+                            d[k] = d[k][::-1]  # flip the data in the array
+
+    def assign_cost_to_graph(self, cost_keys=["hop_distance"]):
+        """
+        Read the data dict on each add and populate a cost key
+        Args:
+            cost_keys: a list of keys for data on each edge.
+                The SC Graph is decorated with a "cost" key that is the product of the different keys here
+        """
+        for k, v in self.unique_hops.items():
+            cost_val = np.prod([v[ik] for ik in cost_keys])
+            self.add_data_to_similar_edges(k, {"cost": cost_val})
+
+    def get_intercollating_path(self, max_val=100000):
+        """
+        obtain an intercollating pathway through the material using hops that are in the current graph
+        Basic idea:
+            Get an endpoint p1 in the graph that is outside the current unit cell
+            Ask the graph for a pathway that connects to p1 from either within the (0,0,0) cell
+            or any other neighboring UC not containing p1.
+        Args:
+            max_val: Filter the graph by a cost
+
+        Returns:
+            Generator for List of Dicts:
+            Each dict contains the information of a hop
+        """
+
+        if len(self.unique_hops) != len(self.unique_hops):
+            logger.error(
+                f"There are {len(self.unique_hops)} SC hops but {len(self.unique_hops)} UC hops in {self.name}"
+            )
+
+        # for u, v, k, d in self.s_graph.graph.edges(data=True, keys=True):
+        for u in self.s_graph.graph.nodes():
+            # for each hop that leave the UC cut that hop and get the path to other cell
+            # if d["to_jimage"] == (0, 0, 0):
+            #     continue
+            # if d["cost"] > max_val:
+            #     continue
+            # Create a copy of the graph that is only composed of internal hops
+            GG = deepcopy(self.s_graph.graph)
```

Comment:
This does not seem to follow python style guides.

Reply:
sorry I clean up the comments around here and gave it a better name too